src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -213,7 +219,9 @@ final case class ProcessApiServiceImpl(
         agreementStates <- parseArrayParameters(agreementStates)
           .traverse(AgreementState.fromValue)
           .toFuture
-        eServices       <-
+        role            <- getUserRolesFuture(contexts)
+        visibilityRestrictions = Seq(ADMIN_ROLE, API_ROLE).contains(role)
```

Comment:
if the requester is a security operator the filter is not applied and the user could see all results, while only Admin and Api operators should see drafts.

Maybe you meant it with a negation?
```suggestion
        visibilityRestrictions = !Seq(ADMIN_ROLE, API_ROLE).contains(role)
```